### ISSUE TITLE
The progress arc of an `outlined` Circular Progress component is slightly off track

### ISSUE THREAD

**Issue by ony3000** [ony3000/tailwind-joy#3]
## Environment information

- Device: iPhone SE (2nd gen.)
- OS: iOS 17.5.1
- Browser: Safari

## Steps to reproduce

1. Go to [the documentation website](https://tailwind-joy.vercel.app).
1. Check out the "Variants" demo on the documentation page for the Circular Progress component.

## The current behavior

![current](https://github.com/user-attachments/assets/61691bbe-e61e-405f-a738-2cb2b571f30f)

## The expected behavior

![expected](https://github.com/user-attachments/assets/75b0d354-c965-4234-a028-0f28eba84b64)

**Comment by ony3000** [comment]
After investigating this issue, I found that the same symptom occurs if any of the ancestors of the Circular Progress component satisfy all of the following conditions:

- Has a style of `display: flex` or `display: grid`.
- Attempts to center the child elements.
- The [parity](https://en.wikipedia.org/wiki/Parity_(mathematics)) of the size of that child element and the size of the content of the element with the display property are different.

Here's an example:

```jsx
{/* odd width */}
<div className="flex w-[75px] justify-center">
  {/* even width */}
  <div className="w-[50px]">
    <CircularProgress variant="outlined" />
  </div>
</div>
```

I also confirmed the same symptom in the Joy UI documentation.
